### PR TITLE
Add `protect_content` to `Defaults`

### LINKED ISSUE #2839: Add Defaults.protected_content …
… since API 5.6 added that parameter to all `send_*` methods.

### PR DESCRIPTION
**Pull request by harshil21**
### Checklist for PRs

- [x] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs

Any more changes to be done?

Closes #2839

### CODE REVIEW

File: telegram/ext/_defaults.py [hunk 1]
```diff
@@ -220,6 +228,21 @@ def run_async(self) -> bool:
...
+        """:obj:`bool`: Optional. Protects the contents of the sent message from forwarding and
+        saving.
+
+        .. versionadded:: 14.0
+        """
+        return self._protect_content
+
+    @protect_content.setter
+    def protect_content(self, value: object) -> NoReturn:
+        raise AttributeError(
```
Comment (Bibo-Joshi) [inline telegram/ext/_defaults.py:242]: codecov says this line wasn't covered

File: telegram/ext/_defaults.py [hunk 2]
```diff
@@ -220,6 +228,21 @@ def run_async(self) -> bool:
...
+            "You can't assign a new value to protect_content after initialization."
+        )
+
     def __hash__(self) -> int:
```
Comment (Bibo-Joshi) [inline telegram/ext/_defaults.py:246]: `__hash__` should be updated + probably tests updates for `__hash__` and `__eq__`

File: tests/test_bot.py [hunk 3]
```diff
@@ -2084,6 +2083,15 @@ def test_send_message_default_parse_mode(self, default_bot, chat_id):
         assert message.text == test_markdown_string
         assert message.text_markdown == escape_markdown(test_markdown_string)
 
+    @flaky(3, 1)
+    @pytest.mark.parametrize('default_bot', [{'protect_content': True}], indirect=True)
+    def test_send_message_default_protect_content(self, default_bot, chat_id):
```
Comment (Bibo-Joshi) [inline tests/test_bot.py:2088]: for the other default settings that we *can* actually test, we do so for all the methods (search for `_default_parse_mode` in the tests). Having more tests is better than having less 😬

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the PR! left some comments. regarding `api_defaults`:

It's public, because we access it outside of `Defaults` in `Bot`. And it doesn't provide any info that's not available otherwise  - you can just check the properties of `Defaults`. In fact, even the properties are of no actual value for a user …
IMO it's not necessary to make it private and then having to shut up pylint about access to private attributes.
If you prefer, we can even just give it a docstring.

File: tests/test_bot.py (hunk 3 above)
Comment (harshil21) [reply tests/test_bot.py:2088]: wow, guess there's a lot of new tests to be written then

Comment (Bibo-Joshi) [reply tests/test_bot.py:2088]: the joys of development 🥳

File: telegram/_message.py [hunk 4]
```diff
@@ -2533,20 +2533,20 @@ def _parse_html(
                 elif entity.type == MessageEntity.URL and urled:
                     insert = f'<a href="{text}">{text}</a>'
                 elif entity.type == MessageEntity.BOLD:
-                    insert = '<b>' + text + '</b>'
+                    insert = f'<b>{text}</b>'
```
Comment (Bibo-Joshi) [inline telegram/_message.py:2536]: can we move these changes to a standalone PR? they arent related to `Defaults.protected_content` … you can probably just cherry-pick the commit(s) to a new branch and revert them on this one

File: tests/test_bot.py [hunk 5]
```diff
@@ -415,7 +414,8 @@ def test_delete_message_old_message(self, bot, chat_id):
     # duplicate here.
 
     @flaky(3, 1)
-    def test_send_venue(self, bot, chat_id):
+    @pytest.mark.parametrize('default_bot', [{'protect_content': True}], indirect=True)
+    def test_send_venue_with_default_protect_content(self, default_bot, chat_id):
```
Comment (Bibo-Joshi) [inline tests/test_bot.py:418]: would you mind extracting the default-testing to a new test? it took a while for me to understand what's going on here and I think having two tests like for the other methods would make it easier

File: tests/test_defaults.py [hunk 6]
```diff
@@ -33,29 +34,14 @@ def test_slot_behaviour(self, mro_slots):
...
-            defaults.timeout = True
-        with pytest.raises(AttributeError):
-            defaults.quote = True
-        with pytest.raises(AttributeError):
-            defaults.tzinfo = True
-        with pytest.raises(AttributeError):
-            defaults.run_async = True
+        for name, val in inspect.getmembers(Defaults, lambda x: isinstance(x, property)):
+            with pytest.raises(AttributeError):
+                setattr(defaults, name, True)
```
Comment (Bibo-Joshi) [inline tests/test_defaults.py:39]: 👍

File: tests/test_updater.py [hunk 7]
```diff
@@ -615,13 +615,13 @@ def test_idle(self, updater, caplog):
...
+            print(idx, log)
             msg = log.getMessage()
             if msg.startswith('Error while getting Updates: Conflict'):
-                caplog.records.pop(idx)  # For stability
+                caplog.records.remove(log)  # For stability
 
-            if msg.startswith('No error handlers are registered'):
-                caplog.records.pop(idx)
+            elif msg.startswith('No error handlers are registered'):
+                caplog.records.remove(log)
```
Comment (Bibo-Joshi) [inline tests/test_updater.py:625]: is this actually fixed or a next iteration of trying to fix it? If it's the first, I'd suggest to again move to a standalone PR

**Review by Bibo-Joshi — COMMENTED** [review]
thanks for the updates!

File: tests/test_updater.py (hunk 7 above)
Comment (harshil21) [reply tests/test_updater.py:625]: hopefully completely fixed now

**Review by Bibo-Joshi — APPROVED** [review]
LGTM! @Poolitzer are you interested in reviewing?

the deepsource failures are unrelated and almost all false positives - let's ignore them on this PR

**Review by Poolitzer — APPROVED** [review]
LGTM!
